### ISSUE TITLE
MA - Received excessive number of results from SR Upload 6/29 and 6/30

### ISSUE THREAD

**Issue by wcollin89** [CDCgov/prime-reportstream#10198]
## **Problem Statement**

Kaleigh Carney submitted a ReportStream ticket via the SS Web Form on 6/30/23 at 11:11 AM EDT regarding an excessively large number of results received from one lab over two days.

## **What you need to know**

Form submission from Kaleigh Carney:

> We received two files (one from 6/29 one 6/30) that were extremely large in size, and look to contain years worth of results. Can someone review this?

On the dates mentioned, the HL7 batches for **ma-phd** include over 20,000+ results each, and the vast majority are associated with one particular lab - "PRL NY Innolabs"

Relevant ReportStream Message IDs: `3621bc67-8b24-45dc-8f0c-946808b11da2`, `bccc957d-0da5-4800-9f73-7a9c095747c9`
Associated CLIAs: `33D2196459`, `21D2215498`, `05D2230382`

`33D2196459` is associated with the largest number of results on each date, both in excess of 20,000 entries. There appears to be a wide range of dates reported on these results as well - I saw April, June, and August of 2022 on a few of the entries in the HL7 file. This CLIA is not associated with any facilities in SimpleReport, though `21D2215498` is associated with "Pandemic Response Lab". 

## **Acceptance Criteria**

- [ ] Determine the cause of the excessive results received by MA PHD.
- [ ] Determine whether any data will need to be corrected and resubmitted by the sender.

## **To Do**

In the Report File table in the RS Metabase, there appear to be duplicates of a certain upload which may be affecting the true total number of results to report.

## **Additional Comments**

- Upon reviewing the upload history for Pandemic Response Lab, the total number of results in their uploads over both days did not equal the smaller total number of entries from the "PRL NY Innolabs" sender on _one_ day.
- Upon reviewing the ReportStream Metabase, it appears that one of the reports is duplicating itself within the batch. There are multiple table entries that all show this report having 5460 results, though it shows up three times for each date. The Blob Digest referenced on all of these duplicate entries is `0xADED8181`.

**Comment by LaFreda1** [comment]
@cloudmagic80 let me know if you don't have the bandwidth to take this. Thanks!

**Comment by wcollin89** [comment]
@LaFreda1 @cloudmagic80 just wanted to check in for any updates - the user has not reached out, I wanted to follow up myself.

Thank you!

**Comment by LaFreda1** [comment]
Hi @wcollin89, I've asked Don to put this on pause. I am discussing this with SR. Will update soon

**Comment by wcollin89** [comment]
@LaFreda1 Thank you very much! I'll keep an eye out for any update requests from the user, but will otherwise wait to reach out until I receive further updates from you or Don.

**Comment by LaFreda1** [comment]
@wcollin89 Update: I heard back from SAR on Friday, I will discuss this ticket with the team during standup this morning and will provide an update. 

SR's message per Merethe on Friday: _if there are certain states that don’t want large uploads or don’t want data older than a certain date, any kind of filtering for that would need to be done in RS since SR doesn’t know who the receivers are for its files, we just send everything to RS and then RS controls the routing
SR is able to add date validations or file size validations on our side but we would need to be sure this is a requirement for all STLTs because it would impact them all. imho SR should be sending anything that is valid data and any filtering related to receiver preferences should be part of the RS pipeline(s)
specifically re: communication, I think the STLTs would need to reach out to the orgs themselves. they are the ones uploading the files and choosing what data to send. the best we could do is try to guide users to guidance provided by the STLTs (if this even always exists?) and let users find the doc for the STLT they anticipate their data will be going to_

**Comment by wcollin89** [comment]
@LaFreda1 thank you for the guidance.

It appears that this organization posted multiple large uploads through 6/30/23, but has not uploaded anything since that evening. I have not yet gotten a response from the sending organization regarding a separate request to correct two entries from these uploads received by WA DOH that should be for different states, but I will reach out to them separately about the reason for the uploads.

**Comment by wcollin89** [comment]
@LaFreda1 I have reached out to the state to inquire about the data filtering. @TaneaY is working with the sending organization on another issue and will address their uploads during their meeting.

Thanks!

**Comment by wcollin89** [comment]
@LaFreda1 @cloudmagic80 per Kaleigh:

> For the report stream feed, we would only want to receive new, present-time results. 

Thank you!

**Comment by LaFreda1** [comment]
@wcollin89 thank you! @cloudmagic80 can we implement such filters for MA?

**Comment by cloudmagic80** [comment]
@wcollin89  Let me check. MA is a receiver, correct?

**Comment by cloudmagic80** [comment]
@wcollin89 I checked the code and also doubled checked with another resources about the dates filtering. There isn't a good way to filter on the dates yet. That would have to be added as a function.

Question. Were some of the results really old? I could see a filter being useful that removes results that are older
cc @LaFreda1

**Comment by wcollin89** [comment]
@cloudmagic80 from what I could see from Kaleigh's request, I think we could probably at least apply a year filter to remove results from prior to the current year. So like an `if year < 2023` type of thing?

**Comment by LaFreda1** [comment]
Plan is to set filters to limit file size

**Comment by LaFreda1** [comment]
Moving this to a medium priority due to UP work

**Comment by cloudmagic80** [comment]
@wcollin89  I don't see report stream message ids: 3621bc67-8b24-45dc-8f0c-946808b11da2 and  bccc957d-0da5-4800-9f73-7a9c095747c9 in prod. I checked report stream and azure. None of them show up. Are they in production?  Also, do  you know what date they were submitted? maybe I can find it that way. cc @LaFreda1

**Comment by cloudmagic80** [comment]
Analyzing both of these 2 large HL7 files ( prime-20230630020241.hl7 and prime-20230630040714.hl7). All the 46k entries are from year 2022. The order control number (ORC) effect date range is consistent from April 2022 to Aug 2022 for all these test entries. But since MA only wants result in 2023, these 2 HL7 files are too old. I will wait on Wendy on what the next steps are.

**Comment by wcollin89** [comment]
@cloudmagic80 I reached out to Kaleigh and they have confirmed that they **do not** want this data resubmitted. Thank you!
